src/convert.cpp

```diff
@@ -1513,7 +1513,7 @@ namespace {
         bool ret = true;
         iconv_t cd;
         cd = iconv_open(to, from);
-        if (cd == (iconv_t)(-1)) {
+        if (!cd) {
```

Comment:
I think the logic has changed here. According to this page:

https://linux.die.net/man/3/iconv_open

iconv_open returns -1 in case of error, which was the value we were checking here. Now you are saying `if (!cd)` that would translate to `if (cd == 0)`, no ?

I would just add the  `static_cast<>()` around it to be totally sure we are not changing logic.

Reply:
static_cast is problematic. Under musl, iconv_t is a void pointer. Under uClibc-ng, it is an int. static_cast will fail in the former case.

Looking at musl's iconv_open, it returns cd when successful, (iconv_t)-1 when it fails. cd cannot evaluate to 0.

The logic is correct.

Reply:
I just wanted to be sure you checked this properly, and it seems so :wink: